### PR TITLE
refactor(checkout): CHECKOUT-6437 Build HostedDropIn component package

### PR DESCRIPTION
**Pull request by bc-peng**
## What?
Refactor the existing `HostedDropIn` component into a shared monorepo package.

## Why?
Lay the foundation work for restructuring `checkout-js` into a monorepo.

## Testing / Proof
- CI checks.

### CODE REVIEW

File: packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    signInCustomer?(): void;
+    validateInstrument?(shouldShowNumberField: boolean): React.ReactNode;
+}
+
+interface HostedDropInPaymentMethodState {
+    isAddingNewCard: boolean;
+    selectedInstrumentId?: string;
+}
+
+interface HostedDropInPaymentMethodHelperMethods {
```
Comment (animesh1987) [inline packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.tsx:63]:
```suggestion
interface HostedDropInPaymentMethodDerivedProps {
```
?

Comment (bc-peng) [reply packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.tsx:63]: I converted HOC into this. Will rename them in this and another PR.

File: packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            language: localeContext.language,
+            onUnhandledError: jest.fn(),
+            paymentForm,
+        };
+
+        defaultProps.method.id = 'digitalriver';
+
+        HostedDropInPaymentMethodTest = (props) => (
+            <CheckoutContext.Provider value={{ checkoutService, checkoutState }}>
+                <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
```
Comment (animesh1987) [inline packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.spec.tsx:85]:
```suggestion
                <LocaleContext.Provider value={localeContext}>
```

**Review by davidchin — APPROVED** [review]
Nice work LGTM!

**Pull request merged**